src/components/avatar-volume-controls.js

```diff
@@ -1,63 +1,103 @@
   },
   remove() {
     APP.gainMultipliers.delete(this.audioEl);
     window.APP.store.removeEventListener("statechanged", this.update);
+    this.playerInfo.el.removeEventListener("remote_mute_updated", this.onRemoteMuteUpdated);
   },
 
-  changeVolumeBy(v) {
-    let gainMultiplier = APP.gainMultipliers.get(this.audioEl);
-    gainMultiplier = THREE.Math.clamp(gainMultiplier + v, 0, MAX_MULTIPLIER);
+  updateGainMultiplier(gainMultiplier, updatePref = false) {
     APP.gainMultipliers.set(this.audioEl, gainMultiplier);
+    const audio = APP.audios.get(this.audioEl);
+    if (audio) {
+      updateAudioSettings(this.audioEl, audio);
+    }
     this.updateVolumeLabel();
+    this.el.emit("gain_multiplier_updated", { gainMultiplier });
+    updatePref && updateAvatarVolumesPref(this.playerInfo.displayName, gainMultiplier, this.isLocalMuted());
+  },
+
+  updateLocalMuted(muted, updatePref = false) {
+    if (muted === true) {
+      APP.mutedState.add(this.audioEl);
+    } else {
+      APP.mutedState.delete(this.audioEl);
+    }
     const audio = APP.audios.get(this.audioEl);
     if (audio) {
       updateAudioSettings(this.audioEl, audio);
     }
+    this.updateVolumeLabel();
+    this.el.emit("local_muted_updated", { muted });
+    updatePref && updateAvatarVolumesPref(this.playerInfo.displayName, this.getGainMultiplier(), this.isLocalMuted());
   },
 
   volumeUp() {
-    const gainMultiplier = APP.gainMultipliers.get(this.audioEl);
-    const step = gainMultiplier > 1 - SMALL_STEP ? BIG_STEP : SMALL_STEP;
-    this.changeVolumeBy(step);
+    let gainMultiplier = APP.gainMultipliers.get(this.audioEl);
+    const step = calcGainStepUp(gainMultiplier);
+    gainMultiplier = THREE.Math.clamp(gainMultiplier + step, 0, MAX_GAIN_MULTIPLIER);
+    this.updateGainMultiplier(gainMultiplier, true);
   },
 
   volumeDown() {
-    const gainMultiplier = APP.gainMultipliers.get(this.audioEl);
-    const step = gainMultiplier > 1 + SMALL_STEP ? BIG_STEP : SMALL_STEP;
-    this.changeVolumeBy(-1 * step);
+    let gainMultiplier = APP.gainMultipliers.get(this.audioEl);
+    const step = -calcGainStepDown(gainMultiplier);
+    gainMultiplier = THREE.Math.clamp(gainMultiplier + step, 0, MAX_GAIN_MULTIPLIER);
+    this.updateGainMultiplier(gainMultiplier, true);
   },
 
   updateVolumeLabel() {
     const gainMultiplier = APP.gainMultipliers.get(this.audioEl);
-    const numBars = Math.min(
-      VOLUME_LABELS.length - 1,
-      gainMultiplier <= 1.001
-        ? Math.floor(gainMultiplier / SMALL_STEP)
-        : Math.floor(VOLUME_LABELS.length / 2 + (gainMultiplier - 1) / BIG_STEP)
+    const numBars = calcLevel(gainMultiplier);
+    this.volumeLabel.setAttribute(
+      "text",
+      "value",
+      gainMultiplier === 0 || this.isLocalMuted() ? "Muted" : VOLUME_LABELS[numBars]
     );
-    this.volumeLabel.setAttribute("text", "value", gainMultiplier === 0 ? "Muted" : VOLUME_LABELS[numBars]);
+  },
+
+  getGainMultiplier() {
```

Comment:
Totally just personal style thing but I would not have made either of these getters as I think the line they are wrapping reads just as if not more clearly.

Reply:
Old habits. Updated